packages/grpc-js-core/src/metadata.ts

```diff
@@ -1,5 +1,8 @@
 import * as http2 from 'http2';
 import {forOwn} from 'lodash';
+const hasOwnProperty = Object.prototype.hasOwnProperty;
```

Comment:
I don't understand the value of caching `hasOwnProperty` like this. If it's trying to solve the problem of some other library modifying `Object.prototype`, this only handles the case where that is modified after this library is loaded.

Reply:
Yea, that's the problem it addresses. It can be useful to prevent tampering by code that is loaded later on. We do it a bit in Node core, but I can rebase it out if you don't want it.

Reply:
I guess I don't see the particular value in preventing tampering after the library is loaded if the prototype could just as easily be tampered with before the library is loaded. In addition, part of what I'm unsure about is why you are particularly worried about that method. What about `RegExp.prototype.test`, or `Array.prototype.map`, or any other built in function that we use?